android/app/src/main/kotlin/com/linagora/android/tmail/MainActivity.kt

```diff
@@ -20,5 +27,37 @@ class MainActivity: FlutterFragmentActivity() {
 
     override fun configureFlutterEngine(flutterEngine: FlutterEngine) {
         GeneratedPluginRegistrant.registerWith(flutterEngine)
+
+        MethodChannel(flutterEngine.dartExecutor.binaryMessenger, NOTIFICATION_GROUP_CHANNEL).setMethodCallHandler { call, result ->
+            try {
+                if (call.method == "getNotificationGroupPermission") {
+                    val groupId = call.arguments
+                    if (groupId is String) {
+                        val isEnabled = getNotificationChannelGroupStatus(applicationContext, groupId)
+                        result.success(isEnabled)
+                    } else {
+                        result.error("Unavailable", "No such group", "")
+                    }
+                } else {
+                    result.notImplemented()
+                }
+            } catch (exception: Throwable) {
+                result.error("UnknownError", exception.message, "")
+            }
+        }
+    }
```

Comment:
do we have a better way to manage code for method channel? what happen if we will have a lot method channel in the future?

Reply:
Separated